core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -90,4 +101,65 @@ protected static boolean getBooleanValueOfVariable( VariableSpace space, String
   protected VariableSpace getSpace( ConnectionDetails connectionDetails ) {
     return connectionDetails.getSpace() == null ? Variables.getADefaultVariableSpace() : connectionDetails.getSpace();
   }
+
+  @Override
+  public boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
+    boolean valid = test( connectionDetails );
```

Comment:
Where is this `test` method implemented? I see it referenced here and in [VFSConnectionProvider](https://github.com/pentaho/pentaho-kettle/pull/9305/files#diff-6e1e5cc18cec64ec90c165cc1a5d5519a21ddf12c960fbbd0120a1051ee9c383R65), but I'm not finding an implementation other than in [OtherConnectionDetailsProvider](https://github.com/pentaho/pentaho-kettle/blob/f4363fec6f52a94e8506c5614d5541706c5dda7a/core/src/main/java/org/pentaho/di/connections/vfs/providers/other/OtherConnectionDetailsProvider.java#L92)

Can you help me better understand this area of code 😅

Reply:
Hi,
The test method is implemented only in BaseVFSConnectionProvider, not in OtherConnectionDetails. 

This method is used to test the VFS connection with root path , to check if the provided root path is valid or not.
1) The verification of VFS credentials is preformed.
2) The checks are performed if the VFS connections supports root path or not.
3) The resolvedRootPath is passed to the static method (getFileObject) which returns fileObject of the provided path. This method does not depend on ConnectionManager or metastore.

Reply:
@singletonc The `test()` methods are implemented in all different VFS Providers. 
Example: LocalProvider https://github.com/pentaho/pdi-plugins-ee/blob/2f37e479ca12ae122f69ff072464e952fb1aadcd/local-vfs/core/src/main/java/com/pentaho/local/provider/LocalProvider.java#L88